contracts/zero-ex/contracts/src/features/NativeOrdersFeature.sol

```diff
@@ -0,0 +1,984 @@
+        // a cancel. It's OK to cancel twice.
+        stor.orderHashToTakerTokenFilledAmount[orderHash] |= HIGH_BIT;
+
+        emit OrderCancelled(orderHash);
+    }
+
+    /// @dev Fill a limit order. Private variant. Does not refund protocol fees.
+    /// @param params Function params.
+    /// @return results Results of the fill.
+    function _fillLimitOrderPrivate(FillLimitOrderPrivateParams memory params)
+        private
+        returns (FillNativeOrderResults memory results)
+    {
+        LibNativeOrder.OrderInfo memory orderInfo = getLimitOrderInfo(params.order);
+
+        // Must be fillable.
+        if (orderInfo.status != LibNativeOrder.OrderStatus.FILLABLE) {
+            LibNativeOrdersRichErrors.OrderNotFillableError(
+                orderInfo.orderHash,
+                uint8(orderInfo.status)
+            ).rrevert();
+        }
+
+        // Must be fillable by the taker.
+        if (params.order.taker != address(0) && params.order.taker != params.taker) {
+            LibNativeOrdersRichErrors.OrderNotFillableByTakerError(
+                orderInfo.orderHash,
+                params.taker,
+                params.order.taker
+            ).rrevert();
+        }
+
+        // Must be fillable by the sender.
+        if (params.order.sender != address(0) && params.order.sender != params.sender) {
+            LibNativeOrdersRichErrors.OrderNotFillableBySenderError(
+                orderInfo.orderHash,
+                params.sender,
+                params.order.sender
+            ).rrevert();
+        }
+
+        // Signature must be valid for the order.
+        {
+            address signer = LibSignature.getSignerOfHash(
+                orderInfo.orderHash,
+                params.signature
+            );
+            if (signer != params.order.maker) {
+                LibNativeOrdersRichErrors.OrderNotSignedByMakerError(
+                    orderInfo.orderHash,
+                    signer,
+                    params.order.maker
+                ).rrevert();
+            }
+        }
+
+        // Pay the protocol fee.
+        results.ethProtocolFeePaid = _collectProtocolFee(params.order.pool);
+
+        // Settle between the maker and taker.
+        (results.takerTokenFilledAmount, results.makerTokenFilledAmount) = _settleOrder(
+            SettleOrderInfo({
+                orderHash: orderInfo.orderHash,
+                maker: params.order.maker,
+                taker: params.taker,
+                makerToken: IERC20TokenV06(params.order.makerToken),
+                takerToken: IERC20TokenV06(params.order.takerToken),
+                makerAmount: params.order.makerAmount,
+                takerAmount: params.order.takerAmount,
+                takerTokenFillAmount: params.takerTokenFillAmount,
+                takerTokenFilledAmount: orderInfo.takerTokenFilledAmount
+            })
+        );
+
+        // Pay the fee recipient.
+        results.takerTokenFeeFilledAmount;
```

Comment:
What is this line doing? (If it's needed, a comment explaining it would be nice.)